### ISSUE TITLE
The refresh token does not automatically refresh correctly when using UseRefreshTokens = true

### ISSUE THREAD

**Issue by giometrix** [auth0/auth0-aspnetcore-authentication#66]
The refresh token does not automatically refresh correctly when using `UseRefreshTokens = true`.

### What was the expected behavior?
Expired token and refresh token get replaced with fresh versions when the access token is expired.

### Reproduction
1. Use refresh tokens (with rotation, but bug exists even when not rotating).  
2. Enable `UseRefreshTokens `
3. When token expires, noticed it does not get updated with a new one.

Stepping through the code, it seems that the issue is that the sdk is not awaiting properly, so that the tokens get replaced _after_ the web request completes (and thus never gets updated).

Specifically, I think the issue is [here](https://github.com/auth0/auth0-aspnetcore-authentication/blob/02a536c15a63682954cec4a16b95e575de48ced1/src/Auth0.AspNetCore.Authentication/Utils.cs#L36).  I think it should be done like the way it is done [here](https://github.com/auth0/auth0-aspnetcore-authentication/blob/df2753bb6e158b0093230bc93b05d874af90ebd2/src/Auth0.AspNetCore.Authentication/OpenIdConnectEventsFactory.cs#L33).

I can have a PR up shortly if it makes your life easier.

### Environment
.net 6.0
Windows
v1.0.1 of `Auth0.AspNetCore.Authentication`

**Comment by giometrix** [comment]
Please see PR #67

**Comment by frederikprijck** [comment]
Should be released now as part of 1.0.2, thanks for the PR!

**Comment by mftcodes** [comment]
> The refresh token does not automatically refresh correctly when using `UseRefreshTokens = true`.

This is still true when using this SDK in the context of a Blazor Server application. I have tried to simplify what I am doing by using the [Blazor Sample App](https://github.com/auth0-samples/auth0-aspnetcore-blazor-server-samples/tree/main), and still no refreshing of tokens occurs. 

Maybe my expectations are wrong... Using the sample app, and setting up my application Id Token Expiration to a low value, say 60 seconds, and the Refresh Token Expiration to similar value, the token's never refresh. I see the same JWT and refresh token after the expiration date. I would expect that while using the app, it would automatically refresh the token at some point, and if I'm idle, I would expect the app to force me to reauthenticate if I try to access the site beyond those expiration times. 

I have also setup an API, to set a shorter Access Token to see if somehow it's expiration time would come into play, and that does not seem to change any behavior either. 

I have noticed, that once I surpass the `Maximum Refresh Token Lifetime`, a hard refresh of the `Profile` page of the sample app continues to give me the original ID and Access tokens, but now the refresh token is blank...

I am assuming, at this point, that I am misunderstanding what the line `options.UseRefreshTokens = true;` is supposed to do. This [blog post](https://auth0.com/blog/use-refresh-tokens-in-aspnet-core-apps/) also seems to indicate that the line should provide some sort of automation of the refresh process, and thinking what needs to occur is an integration of the `Auth0.AuthenticationApi` SDK and manually setup some token refreshing. Which begs the question, what is the point of this SDK if I have to integrate the Authentication API too? 

Anyway, what is this line supposed to actually do within the context of the Blazor Server Sample App? 
```
...
builder.Services.AddAuth0WebAppAuthentication(options =>
{
    options.Domain = builder.Configuration["Auth0:Domain"];
    options.ClientId = builder.Configuration["Auth0:ClientId"];
    options.ClientSecret = builder.Configuration["Auth0:ClientSecret"];
    options.Scope = "openid profile email";
}).WithAccessToken(options =>
{
    options.Audience = builder.Configuration["Auth0:Domain"];
    options.UseRefreshTokens = true;  <--- This is supposed to do what, beyond giving me a refresh token?
});
```
